src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -316,79 +387,216 @@ def main(self):
         if not self.args.bluestore and not self.args.filestore:
             self.args.bluestore = True
 
-        if (self.args.no_auto or self.args.db_devices or
-                                  self.args.journal_devices or
-                                  self.args.wal_devices):
-            self._get_explicit_strategy()
-        else:
-            self._get_strategy()
+        if (not self.args.no_auto and not self.args.db_devices and not
+            self.args.wal_devices and not self.args.journal_devices):
```

Comment:
```suggestion
        if not any([self.args.no_auto,
                    self.args.db_devices,
                    self.args.wal_devices,
                    self.args.journal_devices]):
```
improves readability

Reply:
But harder to reason over imho. In any case the introduction of `--auto` has changed this slightly.